src/gen_rules.ml

```diff
@@ -238,13 +234,29 @@ module Gen(P : Install_rules.Params) = struct
         ~dep_graphs:(Ocamldep.Dep_graphs.dummy m));
 
     if Library.has_stubs lib then begin
+      let all_dirs = Dir_contents.dirs dir_contents in
       let h_files =
-        String.Set.to_list (Dir_contents.text_files dir_contents)
-        |> List.filter_map ~f:(fun fn ->
-          if String.is_suffix fn ~suffix:".h" then
-            Some (Path.relative dir fn)
-          else
-            None)
+        List.fold_left all_dirs ~init:[] ~f:(fun acc dc ->
+          String.Set.fold (Dir_contents.text_files dc) ~init:acc
+            ~f:(fun fn acc ->
+              if String.is_suffix fn ~suffix:".h" then
+                Path.relative (Dir_contents.dir dc) fn :: acc
+              else
+                acc))
+        |> List.rev
```

Comment:
Why do we need this reverse? I just see that we provide these as command line args, does the order matter there?

Reply:
Indeed, it doesn't matter. Note that it's not even command line flags, it's just to specify dependencies